src/FondOfSpryker/Zed/ShipmentDeliveryNoteApi/Business/Model/ShipmentDeliveryNoteApi.php

```diff
@@ -34,88 +27,52 @@ class ShipmentDeliveryNoteApi implements ShipmentDeliveryNoteApiInterface
     }
 
     /**
      * @param \Generated\Shared\Transfer\ApiDataTransfer $apiDataTransfer
      *
+     * @throws \Spryker\Zed\Api\Business\Exception\EntityNotSavedException
+     *
      * @return \Generated\Shared\Transfer\ApiItemTransfer
      */
     public function add(ApiDataTransfer $apiDataTransfer): ApiItemTransfer
     {
         $data = (array)$apiDataTransfer->getData();
 
-        $shipmentDeliveryNoteTransfer = new ShipmentDeliveryNoteTransfer();
-        $shipmentDeliveryNoteTransfer->fromArray($data, true);
+        $shipmentDeliveryNoteTransfer = $this->transferMapper->toTransfer($data);
 
-        $itemsCollection = [];
+        $shipmentDeliveryNoteResponseTransfer = $this->shipmentDeliveryNoteFacade->createShipmentDeliveryNote(
+            $shipmentDeliveryNoteTransfer
+        );
 
-        if (!isset($data['items'])) {
-            $data['items'] = [];
-        }
-
-        foreach ($data['items'] as $item) {
-            $itemsCollection[] = (new ShipmentDeliveryNoteItemTransfer())->fromArray($item, true);
-        }
-
-        $shipmentDeliveryNoteResponseTransfer = $this->shipmentDeliveryNoteFacade->addShipmentDeliveryNote($shipmentDeliveryNoteTransfer, $itemsCollection);
-
-        $shipmentDeliveryNoteTransfer = $this->getShipmentDeliveryNoteFromResponse($shipmentDeliveryNoteResponseTransfer);
-
-        return $this->apiQueryContainer->createApiItem($shipmentDeliveryNoteTransfer, $shipmentDeliveryNoteTransfer->getIdShipmentDeliveryNote());
-    }
-
-    /**
-     * @param \Generated\Shared\Transfer\ShipmentDeliveryNoteResponseTransfer $shipmentDeliveryNoteResponseTransfer
-     *
-     * @return \Generated\Shared\Transfer\ShipmentDeliveryNoteTransfer
-     */
-    protected function getShipmentDeliveryNoteFromResponse(
-        ShipmentDeliveryNoteResponseTransfer $shipmentDeliveryNoteResponseTransfer
-    ): ShipmentDeliveryNoteTransfer
-    {
         $shipmentDeliveryNoteTransfer = $shipmentDeliveryNoteResponseTransfer->getShipmentDeliveryNoteTransfer();
 
-        if (!$shipmentDeliveryNoteTransfer) {
-            $errors = [];
-            foreach ($shipmentDeliveryNoteResponseTransfer->getErrors() as $error) {
-                $errors[] = $error->getMessage();
-            }
-
-            throw new EntityNotSavedException('Could not save Shipment Delivery Note: ' . implode(',', $errors));
+        if ($shipmentDeliveryNoteTransfer === null || $shipmentDeliveryNoteResponseTransfer->getIsSuccess() === false) {
+            throw new EntityNotSavedException(
+                'Could not save shipment delivery note.',
+                ApiConfig::HTTP_CODE_INTERNAL_ERROR
+            );
         }
 
-        return $this->shipmentDeliveryNoteFacade->findShipmentDeliveryNoteById($shipmentDeliveryNoteTransfer);
+        return $this->apiQueryContainer->createApiItem(
+            $shipmentDeliveryNoteTransfer,
+            $shipmentDeliveryNoteTransfer->getIdShipmentDeliveryNote()
+        );
```

Comment:
Is it worth wrapping the creation of the item in an try catch block to actually throw the error when it's actually saved?